From 345d9e6cc85b627d026546f0a9a845a0d7c2bc0a Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Fri, 12 Oct 2018 17:09:08 +0300
Subject: [PATCH 4855/5242] MLK-18789: arch: arm64: fsl-imx8qxp: Fix adma_pwm
 default status

commit  79503e3095473aeb8c3f66cbbc1ddd2bbdf3797d from
https://source.codeaurora.org/external/imx/linux-imx.git

The pwm_adma_lcdif node was created with status "okay" in the mek.dts
file, causing a pin conflict with the audio subsystem, even though the
lcdif pwm is not needed in this dts file.
Move this node into it's specific dts file: mek-lcdif, since it is only
needed by the lcdif.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts   |    2 ++
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |    7 -------
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
index 750d725..4ac48e8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
@@ -81,5 +81,7 @@
 };
 
 &pwm_adma_lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif_pwm>;
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 2d210fd..dc7ee27 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -1200,13 +1200,6 @@
 	status = "okay";
 };
 
-&pwm_adma_lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_pwm>;
-	status = "okay";
-};
-
-
 /* DSI/LVDS port 1 */
 &i2c0_mipi_lvds1 {
 	#address-cells = <1>;
-- 
1.7.9.5

